### ISSUE TITLE
Syntax hangs

### ISSUE THREAD

**Issue by t-bltg** [JuliaEditorSupport/Julia-sublime#100]
Hi, thanks for the work being done here.

I switched to ST4 last weekend, and everything was running smoothly until today.

It seems that https://github.com/JuliaEditorSupport/Julia-sublime/commit/3d47f0c93ab51cddfc8cb825cfdcdf8d6ea47711 makes syntax highlight and the editor hang forever (reverting to `v1.0.2` manually works though).

MWE: open https://github.com/JuliaLang/julia/blob/bb5b98e72a151c41471d8cc14cacb495d647fb7f/base/abstractarray.jl, hangs in a few seconds.

Can anyone reproduce ?

```bash
$ lsb_release -a
LSB Version:	core-11.1.0ubuntu2-noarch:security-11.1.0ubuntu2-noarch
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.2 LTS
Release:	20.04
Codename:	focal
```

**Comment by jaakkor2** [comment]
Yes, v1.0.3 is problematic, v1.0.2 seems to be fine.
If I type on the first line
```
b = (((abs(a))))
```
and try to type after the second opening parenthesis, Sublime hangs.

![image](https://user-images.githubusercontent.com/7318249/119683215-24418e00-be4c-11eb-8de6-6122d38641af.png)

**Comment by randy3k** [comment]
The original fix for #98 was not correct. Please check out v1.0.4

**Comment by t-bltg** [comment]
@randy3k , many thanks !
